pull_api_definition.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+
+# List of required CLI applications
+required_cli_apps=(curl jq sed)
+
+# Output file for the API definition
+output_file=".openapi-hidrivenext-mail-notification.json"
+
+help() {
+	echo "Usage: $0 <host>"
+	echo "  host: The host where the API spec is hosted"
+	echo "  Example: $0 api.example.lan:10443"
+	echo "           export API_SPEC_USER=<user> API_SPEC_PASSWORD=<pass>; ./$0 api.example.lan:10443"
+}
+
+# Check if the host argument is provided
+if [ $# -eq 0 ]; then
+	help
+	exit 1
+fi
+
+# Check if required CLI applications are installed
+for required_cli_apps in "${required_cli_apps[@]}"; do
+	which "${required_cli_apps}" >/dev/null
+	if [ ! $? -eq 0 ]; then
+		echo "[e] ERROR: requirement ${REQ} is missing"
+		exit 1
+	fi
+done
+
+# Set the API spec host and URL
+API_SPEC_HOST=$1
+API_SPEC="https://${API_SPEC_HOST}/easynextcloud-mail-notification/api-docs/NotificationAPI"
+echo "Update API client definition"
+echo "API Spec: ${API_SPEC}"
+
+# Initialize current API version
+current_api_version=""
+
+# Check if the output file exists and get the current API version
+if [ -f "${output_file}" ]; then
+	current_api_version=$(jq -r '.info.version' "${output_file}")
+fi
+
+# Initialize authentication CLI arguments
+auth_cli_args=""
+
+# Set authentication CLI arguments if user and password are provided
+if [ -n "${API_SPEC_USER}" ] && [ -n "${API_SPEC_PASSWORD}" ]; then
+	auth_cli_args="-u $API_SPEC_USER:$API_SPEC_PASSWORD"
+fi
+
+# Create a temporary file to store the downloaded API spec
+temp_file=$(mktemp)
+
+# Download the API spec
+curl --progress-bar "${auth_cli_args}" --insecure "${API_SPEC}" > "${temp_file}"
```

Comment:
Empty auth CLI args lead to an irritating warning. Since basic auth is not required, please drop this:

```diff
@@ -42,19 +42,11 @@ if [ -f "${output_file}" ]; then
        current_api_version=$(jq -r '.info.version' "${output_file}")
 fi
 
-# Initialize authentication CLI arguments
-auth_cli_args=""
-
-# Set authentication CLI arguments if user and password are provided
-if [ -n "${API_SPEC_USER}" ] && [ -n "${API_SPEC_PASSWORD}" ]; then
-       auth_cli_args="-u $API_SPEC_USER:$API_SPEC_PASSWORD"
-fi
-
 # Create a temporary file to store the downloaded API spec
 temp_file=$(mktemp)
 
 # Download the API spec
-curl --progress-bar "${auth_cli_args}" --insecure "${API_SPEC}" > "${temp_file}"
+curl --progress-bar --insecure "${API_SPEC}" > "${temp_file}"
 
 # Check if the download was successful
 if [ ! $? -eq 0 ]; then


```